module/VuFind/src/VuFind/Net/IpAddressUtils.php

```diff
@@ -124,4 +125,34 @@ public function isInRange($ip, $ranges)
         }
         return false;
     }
+
+    /**
+     * Truncate an IP address to the given number of IPv4 octets
+     * or IPv6 hextets, depending what kind of IP address it is.
+     *
+     * @param string $ip          IP address to truncate
+     * @param int    $ipv4Octets  Number of octets to return if it is IPv4
+     * @param int    $ipv6Hextets Number of hextets to return if it is IPv6
+     *
+     * @return string The possibly truncated IP address
+     */
+    public function truncate($ip, $ipv4Octets = null, $ipv6Hextets = null)
+    {
+        if (!str_contains($ip, ':') || !defined('AF_INET6')) {
```

Comment:
It'd be more efficient to check everything here in this order:
`if ($ipv4Octets && (!defined('AF_INET6') || !str_contains($ip, ':'))) {`

Reply:
I agree that would be more efficient, but you don't think it would make the code a little less clear with the expanded elseif? I was thinking instead of refactoring the check for address type, which I copied from a method above, into a helper method isIpv4.

Reply:
True, it's not quite the same. Please disregard!